### PR TITLE
[Paused] Market guide scrolling issue #1583

### PR DESCRIPTION
**Pull request by VerevkinAlexander**
Added method that scrolls central element of Exchange page to highlited.

### CODE REVIEW

**Comment by VerevkinAlexander** [comment]
@sschiessl-bcp Please review it , when you have time.

**Comment by sschiessl-bcp** [comment]
This solution only works for this one component, and also only if there is no help element in the top area of the exchange (because then you would need to scroll up). 

Did you see the comments and hints in the original issue?

**Comment by VerevkinAlexander** [comment]
You mentioned on comments. Did you mean next one? :
-"According to index.js, it works with auto positioning, https://introjs.com/example/auto-position/index.html. Though, I believe that Perfect scrollbar has something to with it not working."
-"Might be enough to update intro.js"

**Comment by sschiessl-bcp** [comment]
> You mentioned on comments. Did you mean next one? :
> -"According to index.js, it works with auto positioning, https://introjs.com/example/auto-position/index.html. Though, I believe that Perfect scrollbar has something to with it not working."
> -"Might be enough to update intro.js"

Yes, those two I am referring to!

**Comment by VerevkinAlexander** [comment]
I tried them but it did not fix our issue. Sorry, that I did not mention you about it.

**Comment by sschiessl-bcp** [comment]
I see. Any fix must be working for any help setup, not just one specific. 

Let's postpone this for now, possible that intro.js needs a fix

File: app/components/Layout/Footer.jsx [hunk 1]
```diff
@@ -108,10 +108,20 @@ class Footer extends React.Component {
 
         var hintData = document.querySelectorAll("[data-intro]");
         var theme = SettingsStore.getState().settings.get("themes");
-        let scrollableDiv = document.getElementsByClassName(
-            "grid-block vertical no-padding ps-container"
+        const scrollableDiv =
+            document.getElementsByClassName(
+                "grid-block vertical no-padding ps-container"
+            )[0] || false;
+        const chartDiv = document.getElementsByClassName(
```
Comment (sschiessl-bcp) [inline app/components/Layout/Footer.jsx:115]: This way of fixing the issue would require a custom solution for *every* panel that uses intro.js.

**Review by sschiessl-bcp — CHANGES_REQUESTED** [review]
The rendering is still broken 

image

image

**Comment by sschiessl-bcp** [comment]
Any solution must be a generic one, and not tailored to a specific panel like the exchange. 

If there is no suitable fix atm, so be it.

**Comment by VerevkinAlexander** [comment]
intro.js scrolling works only for blocks that directly contains in a scrollable block if intro.js hint block is placed in the not scrollable block it needs some additional method for detecting and scrolling main scrollable block. So its need block position calculating for each panel. Also if browser window height is smaller than panel height you will see blank gray block.

**Pull request closed without merging**